src/components/NavBar/NavBar.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import uclaLogo from '../../ucla-samueli-logo-white.png';
+import './NavBar.scss';
+import { BrowserRouter as Router, Route, Link } from "react-router-dom";
+
+class NavBar extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            toggled: false,
+            width: window.innerWidth,
+        };
+    }
+
+    componentDidMount() {
+        this.toggleMenu();
+        window.addEventListener("resize", this.handleWindowSizeChange);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.handleWindowSizeChange);
+    }
+
+    handleWindowSizeChange = () => {
+        this.setState({
+            width: window.innerWidth
+        })
+    };
+
+    toggleMenu = () => {
+        this.setState({
+            toggled: !this.state.toggled
+        }, () => {
+            console.log('state was changed!', this.state.toggled)
+        })
+
+        if (this.state.width <= 760) {
+            if (!this.state.toggled) {
+                document.getElementById("menu-mobile").style.width = "50%";
+            }
+            else {
+                document.getElementById("menu-mobile").style.width = "0";
```

Comment:
For the animation, instead of changing the width of the navbar, change its position by sliding it to the right off the screen. You can do this with CSS `transform: translateX(100%);`.